src/kubernetes.cc

```diff
@@ -762,13 +813,17 @@ void KubernetesReader::WatchMaster(
     // A notification for watch completion.
     std::mutex completion_mutex;
     std::unique_lock<std::mutex> watch_completion(completion_mutex);
-    Watcher watcher(std::bind(&EventCallback, callback, std::placeholders::_1),
+    Watcher watcher(endpoint,
+                    std::bind(&EventCallback, callback, std::placeholders::_1),
                     std::move(watch_completion), config_.VerboseLogging());
     http::client::response response = client.get(request, boost::ref(watcher));
     if (config_.VerboseLogging()) {
       LOG(INFO) << "Waiting for completion";
     }
     std::lock_guard<std::mutex> await_completion(completion_mutex);
+    if (!watcher.exception().empty()) {
+      throw QueryException(watcher.exception());
```

Comment:
Do we really need to throw an exception here? Either way, the thread will end up exiting. Does this exception bubble up to kill the main process?